### ISSUE TITLE
Invalid configuration for layer for height dimension:  Invalid input configuration for kernel height.

### ISSUE THREAD

**Issue by FJGEODEV** [deeplearning4j/deeplearning4j#9169]
#### Issue Description

I have pre-trained a model in Tensoflow 2.2.0 and Keras 2.4.3. It works great in Python enviroment (3.8).
But I can't run it in DL4J, the error message complains the kernel height.

I searched over the internet and found the answer would be "too small image size for the kernel". However, it is not a problem when I used it in Python. 

I also saw DL4J has a "latest version" which is newer version than 1.0.0-beta7.   

Is there any different implementation in Python and Jave to cause this issue or this issue has been fixed in the latest version? 
If later, how can I use maven to add the version into my pom file? Currently I am using:

        <dependency>
            <groupId>org.deeplearning4j</groupId>
            <artifactId>deeplearning4j-core</artifactId>
            <version>${dl4j.version}</version>
        </dependency>

Please describe our issue, along with:
- expected behavior
- encountered behavior

#### Version Information

Please indicate relevant versions, including, if relevant:

* Deeplearning4j version:           1.0.0-beta7
* Platform information (OS, etc) :  Ubuntu (run Python), AWS LInux (DL4J in java)
* CUDA version, if used
* NVIDIA driver version, if in use

#### Additional Information

Where applicable, please also provide:

* Full log or exception stack trace (ideally in a Gist: gist.github.com)
* pom.xml file or similar (also in a Gist)

org.deeplearning4j.exception.DL4JInvalidConfigException: Invalid configuration for layer (idx=-1, name=conv3d_16, type=Convolution3DLayer) for height dimension:  Invalid input configuration for kernel height. Require 0 < kH <= inHeight + 2*padH; got (kH=17, inHeight=16, padH=0)
Input type = InputTypeConvolutional3D(format=NCDHW,d=16,h=16,w=16,c=128), kernel = [3, 3, 3], strides = [1, 1, 1], padding = [0, 0, 0], layer size (output channels) = 128, convolution mode = Same
	at org.deeplearning4j.nn.conf.layers.InputTypeUtil.getOutputTypeCnn3DLayers(InputTypeUtil.java:184)
	at org.deeplearning4j.nn.conf.layers.Convolution3D.getOutputType(Convolution3D.java:122)
	at org.deeplearning4j.nn.modelimport.keras.layers.convolutional.KerasConvolution3D.getOutputType(KerasConvolution3D.java:145)
	at org.deeplearning4j.nn.modelimport.keras.KerasModel.inferOutputTypes(KerasModel.java:304)
	at org.deeplearning4j.nn.modelimport.keras.KerasModel.<init>(KerasModel.java:179)
	at org.deeplearning4j.nn.modelimport.keras.KerasModel.<init>(KerasModel.java:96)
	at org.deeplearning4j.nn.modelimport.keras.utils.KerasModelBuilder.buildModel(KerasModelBuilder.java:307)
	at org.deeplearning4j.nn.modelimport.keras.KerasModelImport.importKerasModelAndWeights(KerasModelImport.java:244)

#### Contributing

Thank you very much!

**Comment by agibsonccc** [comment]
@FanJiangDev  do you have a keras model I could look at?

**Comment by FanJiangDev** [comment]
> @FanJiangDev do you have a keras model I could look at?

Yes, please see the following model:

def UNET1(start_neuron=16, input_size = (128,128,128,1), InputName = 'IMAGE'):
    
    addition = 1
    inputs = Input(input_size, name = InputName)
    
    print('input size =',inputs.shape)
    conv1 = Conv3D(start_neuron*1, (3,3,3), activation = 'relu', padding = 'same')(inputs)
    conv1 = Conv3D(start_neuron*1, (3,3,3), activation = 'relu', padding = 'same')(conv1) 
    pool1 = MaxPooling3D(pool_size=(2, 2, 2))(conv1)
    
    conv2 = Conv3D(start_neuron*2, (3,3,3), activation = 'relu', padding = 'same')(pool1) 
    conv2 = Conv3D(start_neuron*2, (3,3,3), activation = 'relu', padding = 'same')(conv2) 
    pool2 = MaxPooling3D(pool_size=(2, 2, 2))(conv2)
    
    conv3 = Conv3D(start_neuron*4, (3,3,3), activation = 'relu', padding = 'same')(pool2)
    conv3 = Conv3D(start_neuron*4, (3,3,3), activation = 'relu', padding = 'same')(conv3)
    conv3 = Conv3D(start_neuron*4, (3,3,3), activation = 'relu', padding = 'same')(conv3)
    pool3 = MaxPooling3D(pool_size=(2, 2, 2))(conv3)
    
    conv4 = Conv3D(start_neuron*8, (3,3,3), activation = 'relu', padding = 'same')(pool3)
    conv4 = Conv3D(start_neuron*8, (3,3,3), activation = 'relu', padding = 'same')(conv4)
    conv4 = Conv3D(start_neuron*8, (3,3,3), activation = 'relu', padding = 'same')(conv4)
    
    conv4_d2 = Conv3D(start_neuron*8, (3,3,3), activation = 'relu', dilation_rate=2, padding = 'same')(conv4)
    conv4_d2 = Conv3D(start_neuron*8, (3,3,3), activation = 'relu', dilation_rate=2, padding = 'same')(conv4_d2)
    conv4_d2 = Conv3D(start_neuron*8, (3,3,3), activation = 'relu', dilation_rate=2, padding = 'same')(conv4_d2)
    
    dilate1 = Conv3D(start_neuron*8, (3,3,3), activation='relu', padding='same', dilation_rate=1)(conv4_d2)
    dilate2 = Conv3D(start_neuron*8, (3,3,3), activation='relu', padding='same', dilation_rate=2)(dilate1)
    dilate3 = Conv3D(start_neuron*8, (3,3,3), activation='relu', padding='same', dilation_rate=4)(dilate2)
    dilate4 = Conv3D(start_neuron*8, (3,3,3), activation='relu', padding='same', dilation_rate=8)(dilate3)
    dilate5 = Conv3D(start_neuron*8, (3,3,3), activation='relu', padding='same', dilation_rate=16)(dilate4)
    dilate6 = Conv3D(start_neuron*8, (3,3,3), activation='relu', padding='same', dilation_rate=32)(dilate5)
    if addition == 1:
        dilate_all_added = add([dilate1, dilate2, dilate3, dilate4, dilate5, dilate6])
    else:
        dilate_all_added = dilate6
    
    merge5 = concatenate([UpSampling3D(size=(2,2,2))(dilate_all_added),conv3], axis = 4)
    conv5 = Conv3D(start_neuron*4, (3,3,3), activation = 'relu', padding = 'same')(merge5)
    conv5 = Conv3D(start_neuron*4, (3,3,3), activation = 'relu', padding = 'same')(conv5)

    merge6 = concatenate([UpSampling3D(size=(2,2,2))(conv5),conv2], axis = 4)
    conv6 = Conv3D(start_neuron*2, (3,3,3), activation = 'relu', padding = 'same')(merge6)
    conv6 = Conv3D(start_neuron*2, (3,3,3), activation = 'relu', padding = 'same')(conv6)
  
    merge7 = concatenate([UpSampling3D(size=(2,2,2))(conv6),conv1], axis = 4)
    conv7 = Conv3D(start_neuron*1, (3,3,3), activation = 'relu', padding = 'same')(merge7)
    conv7 = Conv3D(start_neuron*1, (3,3,3), activation = 'relu', padding = 'same')(conv7)

    conv8 = Conv3D(1, (1,1,1), activation = 'sigmoid')(conv7)
    print('conv8 size =',conv8.shape)
    model = Model(inputs, conv8)
   
    return model

Thanks for your help.

**Comment by FJGEODEV** [comment]
Hi thanks for your help. I uploaded my model to google drive, please take a look, thanks.

https://drive.google.com/file/d/1QwIU3TgT0hmy08N5tOXCN7Y_M3S8aCPW/view?usp=sharing

**Comment by agibsonccc** [comment]
@FanJiangDev  do you mind giving me access? Feel free to DM me on the forums: https://community.konduit.ai if you need this to be confidential. Thanks!
